src/Go/boardAnalysis/goAI.ts

```diff
@@ -83,35 +94,91 @@ export function makeAIMove(boardState: BoardState, useOfflineCycles = true): Pro
     );
   }
 
-  // Once the AI moves (or the player playing as white with No AI moves),
+  // Once the AI moves (or the player/script playing as white on a "No AI" board),
   // clear the isAiThinking semaphore and update the board UI.
-  Go.nextTurn = Go.nextTurn.finally(() => {
+
+  const playerColor = scriptPlayingAsWhite ? GoColor.white : GoColor.black;
+  Go.nextTurn[playerColor] = Go.nextTurn[playerColor].finally(() => {
     if (!encounteredError) {
       isAiThinking = false;
     }
     GoEvents.emit();
   });
 
-  return Go.nextTurn;
+  return Go.nextTurn[playerColor];
+}
+
+/**
+ * Set up the promises for white and black turns, to handle player / script / AI communication when they are done thinking
+ */
+export function updateTurnPromises() {
+  if (Go.currentGame.previousPlayer === null) {
+    resolveCurrentTurnForWhite();
+    resolveCurrentTurn();
+    Go.nextTurn[GoColor.black] = Promise.resolve({ type: GoPlayType.gameOver, x: null, y: null });
+    Go.nextTurn[GoColor.white] = getNonAIMovePromise(true);
+    return;
+  }
+
+  const previousTurn = getPreviousMove();
+  const previousTurnPromise: Promise<Play> = Promise.resolve(
+    previousTurn
+      ? { type: GoPlayType.move, x: previousTurn[0], y: previousTurn[1] }
+      : { type: GoPlayType.pass, x: null, y: null },
+  );
+
+  if (Go.currentGame.previousPlayer !== GoColor.black) {
```

Comment:
```suggestion
  if (Go.currentGame.previousPlayer === GoColor.white) {
```
IMO it's easier to understand these in positive logic. Otherwise, I wonder what happens if two of them match.